cypress/integration/mocked-okta/signInController.2.cy.ts

```diff
@@ -0,0 +1,135 @@
+import userStatuses from '../../support/okta/userStatuses';
+import ResponseFixture from '../../support/types/ResponseFixture';
+
+beforeEach(() => {
+  cy.mockPurge();
+});
+
+userStatuses.forEach((state) => {
+  context(`Given I am a ${state || 'nonexistent'} user`, () => {
+    context('When I submit the form on /signin', () => {
+      beforeEach(() => {
+        cy.visit(`/signin?useOkta=true`);
+        cy.get('input[name="email"]').type('example@example.com');
+        cy.get('input[name="password"]').type('password');
+      });
+      switch (state) {
+        case false:
+        case 'PROVISIONED':
+        case 'STAGED':
+        case 'RECOVERY':
+        case 'PASSWORD_EXPIRED':
```

Comment:
I think in a few places these tests are kind of misleading. 
For example, here we are generating 4 different tests that all are exactly the same and labelling them as:

>  Given that I am a **[nonexistent/provisioned/staged/recovery/password_expired]** user when I submit the form on /signin Then I should be shown an "Email or password incorrect" error ". 

Unless I'm misunderstanding something this test should only run a single time and be labelled something like 

> 'When I submit the form on /signin' I should be shown an "Email or password incorrect" error if Okta authentication fails' 

Nowhere in the test we are validating that non existent users or users in states other than ACTIVE actually cause Okta to return an authentication failed error.

I would make sure all the tests are labelled in a way that makes it very clear what it is we are verifying. We can add to the test description or comments when we expect to encounter these situations just to get some context.

Reply:
I think for these labels I went with a user-story approach rather than a test case approach - you're probably right in that the test case description would be more useful in this case. That means we could also collapse some identically performing tests down as @coldlink suggested.